spring-websocket/src/main/java/org/springframework/web/socket/adapter/jetty/JettyWebSocketHandlerAdapter.java

```diff
@@ -66,87 +55,89 @@ public JettyWebSocketHandlerAdapter(WebSocketHandler webSocketHandler, JettyWebS
 	}
 
 
-	@OnWebSocketOpen
+	@Override
 	public void onWebSocketOpen(Session session) {
 		try {
+			this.nativeSession = session;
 			this.wsSession.initializeNativeSession(session);
 			this.webSocketHandler.afterConnectionEstablished(this.wsSession);
+			this.nativeSession.demand();
 		}
 		catch (Exception ex) {
-			ExceptionWebSocketHandlerDecorator.tryCloseWithError(this.wsSession, ex, logger);
+			tryCloseWithError(ex);
 		}
 	}
 
-	@OnWebSocketMessage
+	@Override
 	public void onWebSocketText(String payload) {
 		TextMessage message = new TextMessage(payload);
 		try {
 			this.webSocketHandler.handleMessage(this.wsSession, message);
+			this.nativeSession.demand();
 		}
 		catch (Exception ex) {
-			ExceptionWebSocketHandlerDecorator.tryCloseWithError(this.wsSession, ex, logger);
+			tryCloseWithError(ex);
 		}
 	}
 
-	@OnWebSocketMessage
+	@Override
 	public void onWebSocketBinary(ByteBuffer payload, Callback callback) {
-		BinaryMessage message = new BinaryMessage(copyByteBuffer(payload), true);
+		BinaryMessage message = new BinaryMessage(BufferUtil.copy(payload), true);
+		callback.succeed();
 		try {
 			this.webSocketHandler.handleMessage(this.wsSession, message);
-			callback.succeed();
+			this.nativeSession.demand();
 		}
 		catch (Exception ex) {
-			callback.fail(ex);
-			ExceptionWebSocketHandlerDecorator.tryCloseWithError(this.wsSession, ex, logger);
+			tryCloseWithError(ex);
 		}
 	}
 
-	@OnWebSocketFrame
-	public void onWebSocketFrame(Frame frame, Callback callback) {
-		if (OpCode.PONG == frame.getOpCode()) {
-			ByteBuffer payload = frame.getPayload() != null ? frame.getPayload() : EMPTY_PAYLOAD;
-			PongMessage message = new PongMessage(copyByteBuffer(payload));
-			try {
-				this.webSocketHandler.handleMessage(this.wsSession, message);
-				callback.succeed();
-			}
-			catch (Exception ex) {
-				callback.fail(ex);
-				ExceptionWebSocketHandlerDecorator.tryCloseWithError(this.wsSession, ex, logger);
-			}
+	@Override
+	public void onWebSocketPong(ByteBuffer payload) {
```

Comment:
Override for `onWebSocketPing()`?

Reply:
They don't have any logic for WebSocket PING so no point in overriding it.